chat.go

```diff
@@ -13,6 +13,8 @@ const (
 	ChatMessageRoleAssistant = "assistant"
 )
```

Comment:
Please add `function` constants to above line.
`ChatMessageRoleFunction  = "function"`
Refs: https://platform.openai.com/docs/api-reference/chat/create#chat/create-role